src/main/java/org/jenkinsci/plugins/workflow/cps/DSL.java

```diff
@@ -175,6 +195,97 @@ public Object invokeMethod(String name, Object args) {
         }
     }
 
+    private static String loadSoleArgumentKey(StepDescriptor d) {
+        try {
+            String[] names = new ClassDescriptor(d.clazz).loadConstructorParamNames();
+            return names.length == 1 ? names[0] : null;
+        } catch (NoStaplerConstructorException e) {
+            return null;
+        }
+    }
+
+    /**
+     * When {@link #invokeMethod(String, Object)} is calling a generic {@link Descriptor}
+     */
+    protected Object invokeDescribable(Descriptor d, String symbol, Object _args) {
+        StepDescriptor metaStep = findMetaStep(d);
+
+        // The only time a closure is valid is when the resulting Describable is immediately executed via a meta-step
+        NamedArgsAndClosure args = parseArgs(_args, metaStep!=null && metaStep.takesImplicitBlockArgument(), UninstantiatedDescribable.ANONYMOUS_KEY);
+        UninstantiatedDescribable ud = new UninstantiatedDescribable(symbol, null, args.namedArgs);
+
+        if (metaStep==null) {
+            // there's no meta-step associated with it, so this symbol is not executable.
+            // in this case we assume this is building a nested object used as an eventual
+            // parameter of an executable symbol, e.g.,
+            //
+            // hg source: 'https://whatever/', clean: true, browser: kallithea('https://whatever/')
+
+            // also note that in this case 'd' is not trustworthy, as depending on
+            // where this UninstantiatedDescribable is ultimately used, the symbol
+            // might be resolved with a specific type.
+            return ud;
+        } else {
+            try {
+                // execute this Describable through a meta-step
+
+                // split args between MetaStep (represented by mm) and Describable (represented by dm)
+                DescribableModel<?> mm = new DescribableModel(metaStep.clazz);
+                DescribableModel<?> dm = new DescribableModel(d.clazz);
+                DescribableParameter p = mm.getFirstRequiredParameter();
+
+                // order of preference:
+                //      1. mandatory parameter in mm
+                //      2. mandatory parameter in dm
+                //      3. other parameters in mm
+                //      4. other parameters in dm
+                // mm is preferred over dm because that way at least the arguments that dm defines
+                // act consistently
```

Comment:
You mean, the arguments that `dm` defines. Right?

For example, `polling` should always mean the same thing to any call to `checkout`, regardless of any `polling` parameter that the `SCM` might have.